src/figuresClasses.ts

```diff
@@ -1,19 +1,85 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
 
+enum Shape {
+  Triangle = 'triangle',
+  Cicle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
-
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Circle {
+const checkValidFigure = (...sides:number[]): boolean => {
+  return sides.some((side) => side <= 0);
+};
+
+export class Triangle implements Figure {
+  public shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (checkValidFigure(a, b, c)) {
+      throw new Error('Side should be > 0');
+    }
+
+    if (a + b <= c || b + c <= a || c + a <= b) {
+      throw new Error('The longest side is >= than a sum of two others');
+    }
+  }
 
+  getArea(): number {
+    const semiPerimetr = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(
+      semiPerimetr * (semiPerimetr - this.a)
+        * (semiPerimetr - this.b) * (semiPerimetr - this.c),
+    );
+
+    return Math.floor(area * 100) / 100;
```

Comment:
Optional: you round with the same logic twice over the code. You also could create a separate helper function - `rounder` for example for this purpose.